### ISSUE TITLE
Subtitle loading and saving as NuGet packages

### ISSUE THREAD

**Issue by boriseetgerink** [SubtitleEdit/subtitleedit#1971]
At my work we do a lot of automated subtitle conversions using SubtitleEdit. Unfortunately, we currently use .NET's System.Diagnostics.Process to use the command-line version of SubtitleEdit. Instead of that, it would be great if we could use the power of SubtitleEdit, without the need to use the command-line version.

I think this could be done by having a NuGet package that contains the Subtitle, Paragraph and other dependent classes, let's call it the SubtitleEdit.Domain package, for example.

Besides the Domain package, there can be packages for specific subtitle formats that, in turn, use the Domain package (for example: SubtitleEdit.Cavena890 depends on SubtitleEdit.Domain). Each specific subtitle format has a Load and Save method to produce a Subtitle from the Domain and write a Subtitle to a file respectively. The SubtitleEdit program would then use the NuGet packages for the subtitle formats.

@niksedk @ivandrofly @xylographe Are you open to such a change? I think it would open up the SubtitleEdit 'brand' to a larger audience, making it much more usable. I'd like to help with that to come up with a more detailed proposal, if desired.

**Comment by boriseetgerink** [comment]
I've thought about this a bit more: I think the first step would be to move the 'Core' (I think that's a better name than 'Domain') into a separate class library project 'SubtitleEdit.Core' and then move each subtitle format into their own class library project 'Subtitle.Formats.*'. This keeps the source flexible until the organization of classes, interfaces and projects etc. is settled. After that, the projects can be separated more by creating NuGet packages so the app can use them.

**Comment by niksedk** [comment]
We've already separated much of the functionality into 'libse.dll'... but nothing has been fixed for API usage (review access modifiers / comments on public stuff / easy interfaces etc.). Actually especially the subtitle formats need work - and the binary formats have only save-to-file (and not save-to-stream).

Moving each formats into separate packages is probably not a good idea, as they are very small and SE supports way too many formats for this.

It's a good idea with NuGet, but for the next few months I've very little time...
